### DISCUSSION TITLE
Complicated Dynamic Routing

### DISCUSSION THREAD

**Discussion by malikolabiyi** [vercel/next.js#16920] [Help]
Hello, 

I am building a link shortener with NextJS, and having a good experience so far!

But there's an issue I encountered, I want to have website.com/linkToken, but I have some pages with the same routing pattern too, so they conflict and the linkToken gets build each time I visit other pages, this explains better:

```
pages/signup.tsx => website.com/signup
pages/[linkToken].tsx => website.com/HgYwr2

// I don't want to do:
pages/signup.tsx => website.com/signup
pages/p/[linkToken].tsx => website.com/p/HgYwr2
```

So is there any way to exempt some special pages from getting build?

**Comment by alanqchen** [comment]
[Explicit pre-defined routes](https://nextjs.org/docs/routing/dynamic-routes#caveats) will take precedence over the dynamic routes, so this shouldn't be an issue. Maybe there's something else wrong?

**Reply by malikolabiyi** [reply]
Hello, thanks for the response @alanqchen, right, everything worked well, but the issue of building page that gets logged in the console? This means nothing?

**Reply by malikolabiyi** [reply]
Or what's going to happen is that the page doesn't get build again at production, so I won't get this message in the console?

**Reply by alanqchen** [reply]
If I'm understanding you correctly, there's an error when building the page that gets logged? Can you share the message?

**Reply by malikolabiyi** [reply]
No, it's not an error, just like an info. 

So when you go to /signup, two building page info show up:

Building /signup

Buidling /[linkToken]

**Reply by alanqchen** [reply]
Do you use Router.push, Router.write, or any other redirect logic in /signup? It seems that the router correctly builds the correct page but then gets redirected.

**Reply by malikolabiyi** [reply]
No, I didn't do anything related to that... I confirmed it is kind of like treating **/signup as /[linkToken]**, passing **"signup"** as the linkToken here (confirmed by logging the context in the getServerSideProps).

**Reply by alanqchen** [reply]
Is it possible to share the code you have for /signup, it should only show `build page: /[linkToken]` if it truly never reached the actual `/signup` page.

**Reply by malikolabiyi** [reply]
It's a big large, nothing special, just a page rendering a Formik form...

``` tsx
// Signup.tsx
export default function Signup() {
	return (
		<AuthPageLayout title="Create an Account" process="signup">
			<SignupForm />
		</AuthPageLayout>
	);
}
```
```tsx
// [linkToken].tsx
const link = () => {
	return 'Hehe';
};

export const getServerSideProps: GetServerSideProps = async (context) => {
	console.log(context.params);
	return { props: {} };
};
export default link;
```
